Wrappers/Python/cil/optimisation/algorithms/Algorithm.py

```diff
@@ -31,167 +87,140 @@ class Algorithm(object):
+
+        This method triggers :code:`update()` and :code:`update_objective()`
         '''
         if self.should_stop():
-            raise StopIteration()
-        else:
-            if self.iteration == -1 and self.update_objective_interval > 0:
-                self._iteration.append(self.iteration)
-                self.update_objective()
-                self.iteration += 1
-                return
-            time0 = time.time()
-            if not self.configured:
-                raise ValueError('Algorithm not configured correctly. Please run set_up.')
-            self.update()
-            self.timing.append( time.time() - time0 )
+            raise StopIteration
+        if self.iteration == -1 and self.update_objective_interval > 0:
+            self._iteration.append(self.iteration)
+            self.update_objective()
             self.iteration += 1
-            
-            self._update_previous_solution()
-            
-            if self.iteration >= 0 and self.update_objective_interval > 0 and\
-                self.iteration % self.update_objective_interval == 0:
-                
-                self._iteration.append(self.iteration)
-                self.update_objective()
-            
+            return self.iteration
+        if not self.configured:
+            raise ValueError('Algorithm not configured correctly. Please run set_up.')
+        self.update()
+        self.iteration += 1
+
+        self._update_previous_solution()
+
+        if self.iteration >= 0 and self.update_objective_interval > 0 and\
+            self.iteration % self.update_objective_interval == 0:
+
+            self._iteration.append(self.iteration)
+            self.update_objective()
+        return self.iteration
 
     def _update_previous_solution(self):
         """ Update the previous solution with the current one
-        
-        The concrete algorithm calls update_previous_solution. Normally this would 
+
+        The concrete algorithm calls update_previous_solution. Normally this would
         entail the swapping of pointers:
 
         .. highlight:: python
         .. code-block:: python
 
             tmp = self.x_old
             self.x_old = self.x
-            self.x = tmp 
-        
+            self.x = tmp
 
         """
         pass
-        
+
     def get_output(self):
         " Returns the current solution. "
         return self.x
 
-    
     def _provable_convergence_condition(self):
         raise NotImplementedError(" Convergence criterion is not implemented for this algorithm. ")
 
     def is_provably_convergent(self):
         """ Check if the algorithm is convergent based on the provable convergence criterion.
         """
         return self._provable_convergence_condition()
-    
+
     @property
     def solution(self):
         return self.get_output()
-    
-    def get_last_loss(self, **kwargs):
+
+    def get_last_loss(self, return_all=False):
```

Comment:
loss is a term for ML/AI. I used it initially and since then I thought to remove it. Anyone uses it?

Reply:
idk, but naming is worth opening a separate issue for I guess.